..feat Authorization+Guards

Davit-Kobakhidze opened the pull request:
I created authorization and guards tasks and also i modified user class by additional components in user.entity.ts

Davit-Kobakhidze committed:
..feat Authorization+Guards

I created authorization and guards tasks and also i modified user class by additional components in user.entity.ts

1Lazare commented on src/auth/auth.controller.ts:
```diff
@@ -1,21 +1,53 @@
+  }
+
+  @Post('register')
+  async register(@Body() body) {
```
create register dto

1Lazare commented on src/auth/auth.controller.ts:
```diff
@@ -1,21 +1,53 @@
+  @Post('register')
+  async register(@Body() body) {
+    const { username, password, roles } = body;
+    return this.authService.register(username, password, roles);
```
give directly createRegisterDto

1Lazare commented on src/auth/auth.controller.ts:
```diff
@@ -1,21 +1,53 @@
+  constructor(private authService: AuthService) {}
+
+  @Post('login')
+  async login(@Body() body) {
```
same give correct Dto

1Lazare commented on src/auth/auth.controller.ts:
```diff
@@ -1,21 +1,53 @@
+  @Post('login')
+  async login(@Body() body) {
+    const { username, password } = body;
+    const user = await this.authService.validateUser(username, password);
```
write this logic in authService login

1Lazare commented on src/auth/auth.service.ts:
```diff
@@ -1,40 +1,47 @@
 @Injectable()
 export class AuthService {
-  constructor(private readonly usersRepository: UsersRepository) {}
+  @InjectRepository(User)
```
dont injectRepository ,  import userRepository file

1Lazare commented on src/auth/jwt.strategy.ts:
```diff
@@ -0,0 +1,19 @@
+    });
+  }
+
+  async validate(payload: any) {
```
give correct type not any

1Lazare commented on src/users/entities/user.entity.ts:
```diff
@@ -5,31 +5,40 @@ import { Column, CreateDateColumn, DeleteDateColumn, Entity, ManyToMany, OneToMa
-    deleteAt: Date;
-    listeners: any;
+    deleteAt?: Date;
+    listeners?: any;
```
listeners ?

1Lazare replied:
delete you dont need this

1Lazare commented on src/users/entities/user.entity.ts:
```diff
@@ -5,31 +5,40 @@ import { Column, CreateDateColumn, DeleteDateColumn, Entity, ManyToMany, OneToMa
+    listeners?: any;
 
+    @Column('simple-array')
+    roles?: UserRole[];
```
user have one role not array of roles

Davit-Kobakhidze committed:
feat: fixed comments

Fixed all comments except of 1 "roles?: UserRole[];"  if i delete "[]"  i got error in auth.service.ts and can't fix it

Davit-Kobakhidze committed:
feat: Finished Auth Module

fixed all errors and finished the task
